packages/react-scripts/config/webpack.config.js

```diff
@@ -623,9 +623,11 @@ module.exports = function(webpackEnv) {
           navigateFallbackBlacklist: [
             // Exclude URLs starting with /_, as they're likely an API call
             new RegExp('^/_'),
-            // Exclude URLs containing a dot, as they're likely a resource in
-            // public/ and not a SPA route
-            new RegExp('/[^/]+\\.[^/]+$'),
+            // Exclude any URLs whose last part seems to be a file extension
+            // as they're likely a resource and not a SPA route.
+            // URLs containing a "?" character won't be blacklisted as they're likely
+            // a route with query params (e.g. auth callbacks).
+            new RegExp('/[^/?]+.[^/]+$'),
```

Comment:
The `.` needs to be `\\.`, since you're trying to match a literal `.` character (not a wildcard).

Reply:
LoL sorry! I tested it with all the special chars escaped (without regexp constructor), but then I pasted it in the code and removed all the escapes... 😅Good catch